src/components/AccountDrawer/MiniPortfolio/Activity/ActivityRow.tsx

```diff
@@ -59,6 +60,8 @@ export function ActivityRow({ activity }: { activity: Activity }) {
     window.open(getExplorerLink(chainId, hash, ExplorerDataType.TRANSACTION), '_blank')
   }, [offchainOrderStatus, chainId, hash, openOffchainActivityModal])
 
+  if (isSpam) return null
```

Comment:
can we move this check to the parent component instead? to avoid doing all these calls for a row we not actually gonna show

i.e. 
```
{activityGroup.transactions.map((activity) => (!activity.isSpam &&
                <ActivityRow key={activity.hash} activity={activity} />
              ))}
```

Reply:
➕

Reply:
there was no eng intuition here I think I just forgot how to react. great callout will do ty